src/strucclustutils/structcreatedb.cpp

```diff
@@ -89,117 +91,331 @@ std::string removeModel(const std::string& input) {
     std::string suffix = input.substr(secondUnderscoreIndex+1);
     return prefix + suffix;
 }
+
+void addMissingAtomsInStructure(GemmiWrapper &readStructure, PulchraWrapper &pulchra){
+    for(size_t ch = 0; ch < readStructure.chain.size(); ch++){
+        size_t chainStart = readStructure.chain[ch].first;
+        size_t chainLen = readStructure.chain[ch].second - readStructure.chain[ch].first;
+        // Detect if structure is Ca only
+        if (std::isnan(readStructure.n[chainStart + 0].x) &&
+            std::isnan(readStructure.n[chainStart + 1].x) &&
+            std::isnan(readStructure.n[chainStart + 2].x) &&
+            std::isnan(readStructure.n[chainStart + 3].x) &&
+            std::isnan(readStructure.c[chainStart + 0].x) &&
+            std::isnan(readStructure.c[chainStart + 1].x) &&
+            std::isnan(readStructure.c[chainStart + 2].x) &&
+            std::isnan(readStructure.c[chainStart + 3].x)) {
+            pulchra.rebuildBackbone(&readStructure.ca[chainStart],
+                                    &readStructure.n[chainStart],
+                                    &readStructure.c[chainStart],
+                                    &readStructure.ami[chainStart],
+                                    chainLen);
+        }
+    }
+}
+
+void findInterfaceResidues(GemmiWrapper &readStructure, std::pair<size_t, size_t> res1, std::pair<size_t, size_t> res2,
+                           std::vector<size_t> & resIdx1)
+{
+    std::vector<Vec3> coord1, coord2;
+    LocalParameters& par = LocalParameters::getLocalInstance();
```

Comment:
Please pass in the parameters as function parameters, don't request the parameter singleton so deep in a function

Reply:
yep